tests/functional/test_event_alias.py

```diff
@@ -0,0 +1,612 @@
+        "endpoint_prefix": "secretsmanager",
+        "service_id": "secrets-manager"
+    },
+    "serverlessrepo": {
+        "endpoint_prefix": "serverlessrepo",
+        "service_id": "serverlessapplicationrepository"
+    },
+    "servicecatalog": {
+        "endpoint_prefix": "servicecatalog",
+        "service_id": "service-catalog"
+    },
+    "servicediscovery": {
+        "endpoint_prefix": "servicediscovery",
+        "service_id": "servicediscovery"
+    },
+    "ses": {
+        "endpoint_prefix": "email",
+        "service_id": "ses"
+    },
+    "shield": {
+        "endpoint_prefix": "shield",
+        "service_id": "shield"
+    },
+    "sms": {
+        "endpoint_prefix": "sms",
+        "service_id": "sms"
+    },
+    "snowball": {
+        "endpoint_prefix": "snowball",
+        "service_id": "snowball"
+    },
+    "sns": {
+        "endpoint_prefix": "sns",
+        "service_id": "sns"
+    },
+    "sqs": {
+        "endpoint_prefix": "sqs",
+        "service_id": "sqs"
+    },
+    "ssm": {
+        "endpoint_prefix": "ssm",
+        "service_id": "ssm"
+    },
+    "stepfunctions": {
+        "endpoint_prefix": "states",
+        "service_id": "sfn"
+    },
+    "storagegateway": {
+        "endpoint_prefix": "storagegateway",
+        "service_id": "storage-gateway"
+    },
+    "sts": {
+        "endpoint_prefix": "sts",
+        "service_id": "sts"
+    },
+    "support": {
+        "endpoint_prefix": "support",
+        "service_id": "support"
+    },
+    "swf": {
+        "endpoint_prefix": "swf",
+        "service_id": "swf"
+    },
+    "transcribe": {
+        "endpoint_prefix": "transcribe",
+        "service_id": "transcribe"
+    },
+    "translate": {
+        "endpoint_prefix": "translate",
+        "service_id": "translate"
+    },
+    "waf": {
+        "endpoint_prefix": "waf",
+        "service_id": "waf"
+    },
+    "waf-regional": {
+        "endpoint_prefix": "waf-regional",
+        "service_id": "waf-regional"
+    },
+    "workdocs": {
+        "endpoint_prefix": "workdocs",
+        "service_id": "workdocs"
+    },
+    "workmail": {
+        "endpoint_prefix": "workmail",
+        "service_id": "workmail"
+    },
+    "workspaces": {
+        "endpoint_prefix": "workspaces",
+        "service_id": "workspaces"
+    },
+    "xray": {
+        "endpoint_prefix": "xray",
+        "service_id": "xray"
+    }
+}
+
+
+def test_event_alias():
+    for client_name in SERVICES.keys():
+        endpoint_prefix = SERVICES[client_name]['endpoint_prefix']
+        service_id = SERVICES[client_name]['service_id']
+        yield _assert_handler_called, client_name, endpoint_prefix
+        yield _assert_handler_called, client_name, service_id
+
+
+def _assert_handler_called(client_name, event_part):
+    hook_calls = []
+
+    def _hook(**kwargs):
+        hook_calls.append(kwargs['event_name'])
+
+    session = _get_session()
+    session.register('creating-client-class.%s' % event_part, _hook)
```

Comment:
So I was thinking we would have two separate tests:

* `test_event_alias_service_name`: This would use the `create-client-class` event and we would do the `_assert_handler_called` for both `service_name` and `service_id`

* `test_event_alias_endpoint_prefix`: This would use one of the events that was based on endpoint prefexies and we would do the `_assert_handler_called` for both `endpoint_prefix` and `service_id`

The main reasons that I suggest this is that:
1) We are not testing any of the aliasing for service names
2) It is not representative to register on the `endpoint_prefix` for the `creating-client-class` since it used to rely on the `service_name` before.

Reply:
We talked offline about this. I've added another yield to test registering with the client name, and registering against an event that used to be endpoint_prefix is covered by the sagemaker tests (since there's no generic place to test that).